bootstrap/kubeadm/types/upstreamv1beta2/types.go

```diff
@@ -126,6 +126,7 @@ type ClusterConfiguration struct {
 	//
 	// Deprecated: As hyperkube is itself deprecated, this fields is too. It will be removed in future kubeadm config versions, kubeadm
 	// will print multiple warnings when set to true, and at some point it may become ignored.
+	// TODO: Remove this field once conversions to Kubeadm v1beta2 --> v1alpha4 --> v1beta2 are no longer supported.
 	// +optional
```

Comment:
I'm not sure this TODO is helpful. 

We will drop the whole package at some point (when we don't support the Kubernetes/kubeadm versions which require kubeadm v1beta2 types anymore)

Similar to this issue for v1beta1: https://github.com/kubernetes-sigs/cluster-api/issues/4507

Reply:
Agreed - I'll revert the kubeadm types too